hera_sim/tests/test_foregrounds.py

```diff
@@ -1,220 +1,138 @@
+@pytest.fixture(scope="function")
+def freqs():
+    return np.linspace(0.1, 0.2, 100, endpoint=False)
 
-        # simulate the effect
-        vis = foregrounds.diffuse_foreground(
-            lsts,
-            freqs,
-            bl_vec,
-            Tsky_mdl=Tsky_mdl,
-            omega_p=omega_p,
-            delay_filter_kwargs=delay_filter_kwargs,
-            fringe_filter_kwargs=fringe_filter_kwargs,
-        )
 
-        # check the shape
-        self.assertEqual(vis.shape, (lsts.size, freqs.size))
+@pytest.fixture(scope="function")
+def lsts():
+    return np.linspace(0, 2 * np.pi, 1000)
 
-        # check that an autocorrelation is real-valued
-        self.assertTrue(np.all(vis.imag == 0))
 
-        # any other tests we can do?
-
-        # import uvtools, pylab as plt
-        # uvtools.plot.waterfall(vis, mode='log'); plt.colorbar(); plt.show()
-
-        # throw a value error if Tsky_mdl or omega_p not specified
-        nt.assert_raises(
-            ValueError,
-            foregrounds.diffuse_foreground,
-            lsts,
-            freqs,
-            bl_vec,
-            Tsky_mdl=None,
-        )
-        nt.assert_raises(
-            ValueError,
-            foregrounds.diffuse_foreground,
-            lsts,
-            freqs,
-            bl_vec,
-            Tsky_mdl=Tsky_mdl,
-            omega_p=None,
-        )
+@pytest.fixture(scope="function")
+def Tsky_mdl():
+    datafile = DATA_PATH / "HERA_Tsky_Reformatted.npz"
+    return Tsky(datafile)
 
-    def test_pntsrc_foreground(self):
-        # make some parameters
-        freqs = np.linspace(0.1, 0.2, 100, endpoint=False)
-        lsts = np.linspace(0, 2 * np.pi, 1000)
-        bl_vec = [300.0, 0]
 
-        # simulate the effect
-        vis = foregrounds.pntsrc_foreground(lsts, freqs, bl_vec, nsrcs=200)
-
-        # check the shape
-        self.assertEqual(vis.shape, (lsts.size, freqs.size))
-
-        # check that an autocorrelation is real-valued
-        bl_vec = [0, 0, 0]
-        vis = foregrounds.pntsrc_foreground(lsts, freqs, bl_vec, nsrcs=200)
-
-        np.testing.assert_allclose(vis.imag, 0, atol=1e-8)
-
-        # XXX check more substantial things
-        # import uvtools, pylab as plt
-        # uvtools.plot.waterfall(vis, mode='phs'); plt.colorbar(); plt.show()
-
-    def test_diffuse_foreground_orientation(self):
-        # make some parameters
-        freqs = np.linspace(0.1, 0.2, 100, endpoint=False)
-        omega_p = Beam(beamfile)
-        lsts = np.linspace(0, 2 * np.pi, 1000)
-        Tsky_mdl = noise.HERA_Tsky_mdl["xx"]
-        bl_vec = (0, 30.0)
-        fringe_filter_kwargs = {"fringe_filter_type": "tophat"}
-
-        # simulate the effect
+@pytest.mark.parametrize("model", ["pntsrc", "diffuse"])
+def test_foreground_shape(freqs, lsts, Tsky_mdl, omega_p, model):
+    bl_vec = [0, 0, 0]
+    if model == "pntsrc":
+        vis = foregrounds.pntsrc_foreground(lsts=lsts, freqs=freqs, bl_vec=bl_vec)
```

Comment:
would it make sense to make pntsrc and diffuse vis fixtures? You use them in two different tests. Could scope them as "module" to make tests faster, since they're read-only.

Reply:
eh, not really. maybe for the simple tests (shape testing and reality testing for autos), since those use the same set of visibilities? otherwise I think it would be a lot more of a headache to modify the tests to jive with making the foregrounds fixtures. the main issue I have here is that the tests on the crosses don't lend themselves well to reusing the visibilities: we only have two tests on the crosses as of now, and one of them requires simulating visibilities for two orientations (the given baseline and its conjugate) in the same test.